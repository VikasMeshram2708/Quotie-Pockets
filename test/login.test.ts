import { describe, it, expect } from "vitest";

describe("login route", () => {
  it("when successfull logged in", async () => {
    const testEmail = "test11@gmail.com";
    const testPassword = "tests";
    const response = await fetch("http://localhost:5000/api/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        email: testEmail,
        password: testPassword,
      }),
    });
    const result = await response.json();
    expect(result.success).toBe(true);
    expect(result).toMatchSnapshot();
  });

  it("when logged in failed", async () => {
    const testEmail = "test1@gmail.com";
    const testPassword = "tests";
    const response = await fetch("http://localhost:5000/api/login", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        email: testEmail,
        password: testPassword,
      }),
    });
    const result = await response.json();
    expect(result.success).toBe(false);
    expect(result).toMatchSnapshot();
  });
});
