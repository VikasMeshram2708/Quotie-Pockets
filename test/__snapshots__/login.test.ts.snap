// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`login route > when logged in failed 1`] = `
{
  "message": "Unauthorized User.",
  "success": false,
}
`;

exports[`login route > when successfull logged in 1`] = `
{
  "message": "Login successful.",
  "success": true,
  "user": {
    "userEmail": "test11@gmail.com",
    "userId": "65e2b9820d04837dfc58f1a9",
    "userName": "test",
  },
}
`;
